SixTrack/tools/astuce/astuce++.cpp

```diff
@@ -0,0 +1,708 @@
+	}
+}
+
+bool Astuce::ReplaceCA(std::map<std::string, std::list<LineStorage> >::iterator Storage_itr)
+{
+	//Storage_itr = Each block
+	//Lines_itr = Each line in the block
+
+	//CallStorage_itr A selected global block to insert
+	//Insert_itr Each line in the global block
+
+	bool Replaced = false;
+
+	std::string BlockName = Storage_itr->first;
+	
+	std::list<LineStorage>::iterator Lines_itr = Storage_itr->second.begin();
+	while(Lines_itr != Storage_itr->second.end())
+	{
+		if(Lines_itr->Enabled == true)
+		{
+			if(Lines_itr->Text.substr(0,3) == "+ca")
+			{
+				//Get the +ca name
+				size_t NameStart = Lines_itr->Text.find_first_not_of(" ", 4);
+				size_t NameEnd = Lines_itr->Text.find_first_of(" ", NameStart+1);
+				std::string FlagName = FixCase(Lines_itr->Text.substr(NameStart,NameEnd-NameStart));
+
+				//Find this name
+				std::map<std::string, std::list<LineStorage> >::iterator CallStorage_itr = CallStorage.begin();
+				CallStorage_itr = CallStorage.find(FlagName);
+				if(CallStorage_itr != CallStorage.end())
+				{
+					//Delete the +ca statement line
+					Lines_itr = Storage_itr->second.erase(Lines_itr);
+
+					LineStorage Title;
+					Title.Enabled=true;
+					Title.Text="!!! START " + FlagName;
+///					Storage_itr->second.insert(Lines_itr,Title);
+
+					std::list<LineStorage>::iterator Insert_itr = CallStorage_itr->second.begin();
+					while(Insert_itr != CallStorage_itr->second.end())
+					{
+						//Do the insert
+						Storage_itr->second.insert(Lines_itr, *Insert_itr);
+
+						//Advance to the next line
+						Insert_itr++;
+					}
+					Replaced = true;
+
+
+					Title.Text="!!! END " + FlagName;
+///					Storage_itr->second.insert(Lines_itr,Title);
+				}
+				else
+				{
+					std::cerr << "Searched for " << BlockName << " and it was not defined anywhere!" << std::endl;
+					std::cerr << "The following +cd blocks exist:" << std::endl;
+					CallStorage_itr = CallStorage.begin();
+					while(CallStorage_itr != CallStorage.end())
+					{
+						std::cerr << "\"" << CallStorage_itr->first << "\" ";
+						CallStorage_itr++;
+					}
+					std::cerr << std::endl;
+
+					std::cerr << "Calls entry count: " << CallStorage.size() << std::endl;
+					std::cerr << "Decks entry count: " << DeckStorage.size() << std::endl;
+
+					exit(EXIT_FAILURE);
+				}
+
+				Lines_itr--;
+			}
+		}
+		Lines_itr++;
+	}
+	return Replaced;
+}
+
+void Astuce::DoOutput()
+{
+	std::set<std::string>::const_reverse_iterator Decks_itr = Decks.rbegin();
+	while(Decks_itr != Decks.rend())
+	{
+		std::map<std::string, std::list<LineStorage> >::iterator DeckStorage_itr = DeckStorage.begin();
+
+		DeckStorage_itr = DeckStorage.find(*Decks_itr);
+		if(DeckStorage_itr != DeckStorage.end())
+		{
+			std::list<LineStorage>::iterator Lines_itr = DeckStorage_itr->second.begin();
+			while(Lines_itr != DeckStorage_itr->second.end())
+			{
+				if(Lines_itr->Enabled == true)
+				{
+					OutputFileBufferStream << Lines_itr->Text.substr(0,80) << "\n";
```

Comment:
Put a comment that you are cropping to 80 chars.

Eventually, disabling this could be a keyword in the .ast file...